src/mission_companion.cpp

```diff
@@ -0,0 +1,1385 @@
+        caravan_return(p, "evac_center_18", "_commune_refugee_caravan");
+    }
+
+    if (cur_key == "Purchase East Field"){
+        field_build_1(p);
+    }
+    if (cur_key == "Upgrade East Field I"){
+        field_build_2(p);
+    }
+    if (cur_key == "Plant East Field"){
+        field_plant(p, "ranch_camp_63");
+    }
+    if (cur_key == "Harvest East Field"){
+        field_harvest(p, "ranch_camp_63");
+    }
+    if (cur_key == "Assign Scavenging Patrol"){
+        individual_mission(p, "departs on the scavenging patrol...", "_scavenging_patrol");
+    }
+    if (cur_key == "Retrieve Scavenging Patrol"){
+        scavenging_patrol_return(p);
+    }
+    if (cur_key == "Assign Scavenging Raid"){
+        individual_mission(p, "departs on the scavenging raid...", "_scavenging_raid");
+    }
+    if (cur_key == "Retrieve Scavenging Raid"){
+        scavenging_raid_return(p);
+    }
+    if (cur_key == "Assign Ally to Menial Labor"){
+        individual_mission(p, "departs to work as a laborer...", "_labor");
+    }
+    if (cur_key == "Recover Ally from Menial Labor"){
+        labor_return(p);
+    }
+    if (cur_key == "Assign Ally to Carpentry Work"){
+        individual_mission(p, "departs to work as a carpenter...", "_carpenter");
+    }
+    if (cur_key == "Recover Ally from Carpentry Work"){
+        carpenter_return(p);
+    }
+    if (cur_key == "Assign Ally to Forage for Food"){
+        individual_mission(p, "departs to forage for food...", "_forage");
+    }
+    if (cur_key == "Recover Ally from Foraging"){
+        forage_return(p);
+    }
+
+    return true;
+}
+
+void talk_function::individual_mission(npc *p, std::string desc, std::string id, bool group)
+{
+    npc *comp = companion_choose();
+    if (comp == NULL){
+        return;
+    }
+    popup("%s %s", comp->name.c_str(), desc.c_str());
+    comp->companion_mission = p->name + id;
+    if (group){
+        comp->companion_mission_time = -1;
+    } else {
+        comp->companion_mission_time = calendar::turn.get_turn();
+    }
+    companion_leave(comp);
+}
+
+void talk_function::caravan_depart(npc *p, std::string dest, std::string id)
+{
+    std::vector<npc *> npc_list = companion_list(p->name+id);
+    int distance = caravan_dist(dest);
+    int time = 200 + distance * 100;
+    popup("The caravan departs with an estimated total travel time of %d hours...", int(time/600));
+
+    for( auto &elem : npc_list ) {
+        if (elem->companion_mission_time == -1){
+            //Adds a 10% error in estimated travel time
+            elem->companion_mission_time = time + int(time*rng(-.1,.1)) + calendar::turn.get_turn();
+        }
+    }
+
+}
+
+//Could be expanded to actually path to the site, just returns the distance
+int talk_function::caravan_dist(std::string dest)
+{
+    const tripoint site = overmap_buffer.find_closest( g->u.global_omt_location(), dest, 0, false );
+    int distance = sqrt(pow(site.x-g->u.posx(),2)+pow(site.y-g->u.posy(),2));
```

Comment:
rl_dist() should do the right thing, circular vs square distance should apply everywhere.